### PR TITLE
Add KeyOrValue for styling purposes

### PR DESCRIPTION
**Pull request by futurepaul**
Here's a proof of concept of `KeyOrValue`, which is sort of the logical conclusion of #526 .

When providing style values to a widget, the user can use a named value from the `Env`, which will be resolved later by the widget, or they can just use a "concrete" value. I've only implemented this for `Container`, as a proof of concept, but I believe this should work for all widgets that accept an `impl ValueType` in their constructor.

I'm not sure of the performance implications, especially with how much cloning I'm doing. Also, this is with `Key` still taking a `'static str`, lifetimes get really hairy when `Key` isn't static.

The other limitation is that this currently doesn't work for gradients.

### CODE REVIEW

File: druid/src/widget/container.rs [hunk 1]
```diff
@@ -16,18 +16,18 @@
...
-    width: f64,
-    brush: PaintBrush,
+    width: KeyOrValue<f64>,
+    brush: KeyOrValue<Color>,
 }
 
 /// A widget that provides simple visual styling options to a child.
 pub struct Container<T> {
-    background: Option<PaintBrush>,
+    background: Option<KeyOrValue<Color>>,
```
Comment (cmyr) [inline druid/src/widget/container.rs]: I think this isn't quite right, but is useful as an illustration.

We'll still likely want a `PaintBrush` enum in druid proper that includes `KeyOrValue<Color>` but perhaps some other things as well.

File: druid/examples/panels.rs [hunk 2]
```diff
@@ -15,25 +15,23 @@
...
 
 fn build_app() -> impl Widget<()> {
     let solid = Color::rgb8(0x3a, 0x3a, 0x3a);
-    let gradient = LinearGradient::new(
-        UnitPoint::TOP_LEFT,
-        UnitPoint::BOTTOM_RIGHT,
-        (Color::rgb8(0x11, 0x11, 0x11), Color::rgb8(0xbb, 0xbb, 0xbb)),
-    );
+
+    // let gradient = LinearGradient::new(
```
Comment (cmyr) [inline druid/examples/panels.rs]: we can get rid of this file too I think, then looks good to me!

File: druid/src/env.rs [hunk 3]
```diff
@@ -76,6 +78,18 @@ pub enum Value {
     String(String),
 }
 
+/// Either a concrete `T` or a [`Key<T>`] that can be resolved in the [`Env`].
+///
+/// This is a way to allow widgets to interchangeably use either a specific
+/// value or a value from the environment for some purpose.
+///
+/// [`Key<T>`]: struct.Key.html
+/// [`Env<T>`]: struct.Env.html
```
Comment (cmyr) [inline druid/src/env.rs]: oops my mistake, this is wrong, should just be `Env`.

Comment (cmyr) [inline druid/src/env.rs]:
```suggestion
/// [`Env`]: struct.Env.html
```

Comment (futurepaul) [reply druid/src/env.rs]: not sure what you're referring to

**Pull request merged**